src/wui/load_or_save_game.cc

```diff
@@ -135,39 +122,41 @@ bool LoadOrSaveGame::compare_map_name(uint32_t rowa, uint32_t rowb) const {
 std::unique_ptr<SavegameData> LoadOrSaveGame::entry_selected() {
 	std::unique_ptr<SavegameData> result(new SavegameData());
 	size_t selections = table_->selections().size();
-	if (selections == 1) {
+	set_tooltips_of_buttons(selections);
+
+	const std::vector<SavegameData> savegames = get_selected_savegames();
+	game_details_.display(savegames);
+	if (selections > 0 && !selection_contains_directory()) {
+		delete_->set_enabled(true);
+	} else {
+		delete_->set_enabled(false);
+		delete_->set_tooltip("");
+	}
+	// TODO(GunChleoc): Take care of the OK button too.
+	if (selections > 0) {
```

Comment:
You're already checking for emptiness above, so you can restructure this to save a check.

Reply:
I don't think so - or I just don't get it.
Above is a "logical and" check. So to avoid checking "selections > 0" twice I would need to make the first if combined statement a nested one, right? So no check saved but nested - which I personally do not like at all...

Reply:
I prefer the nested. If we have selections, do something, and directories then get special treatment.

Alternatively, you can save some nesting by using an early return:

```
if (table_->selections().empty() {
    return result;
}
// Do something
return result;
```

Reply:
yeah, I updated a slightly different solution. Looks better now, right? :)

Reply:
Yep. I like it :)